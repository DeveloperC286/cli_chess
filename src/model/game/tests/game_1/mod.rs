use super::*;

#[test]
fn test_game_1() {
    // Theodor von Scheve vs Richard Teichmann, Berlin 1907
    // 1. e4  e5  2. Nf3  Nc6  3. Bc4 Bc5 4. c3 Qe7 5. O-O d6 6. d4 Bb6
    // 7. a4 a6 8. a5 Ba7 9. h3 Nf6 10. dxe5 Nxe5 11. Nxe5 Qxe5 12. Nd2 Bxh3
    // 13. gxh3 Qg3+ 14. Kh1 Qxh3+ 15. Kg1 Ng4 16. Nf3 Qg3+ 17. Kh1 Bxf2

    let mut game = crate::model::game::Game::new();

    let movement = Movement::from("e4").unwrap();
    game.move_piece(movement).unwrap();
    assert_debug_snapshot!("1w", game.get_board_representation());

    let movement = Movement::from("e5").unwrap();
    game.move_piece(movement).unwrap();
    assert_debug_snapshot!("1b", game.get_board_representation());
}
