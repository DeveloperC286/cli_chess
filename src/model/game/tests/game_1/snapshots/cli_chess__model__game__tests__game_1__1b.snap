---
source: src/model/game/tests/game_1/mod.rs
expression: game.get_board_representation()

---
[
    " r n b q k b n r",
    " p p p p - p p p",
    " - - - - - - - -",
    " - - - - p - - -",
    " - - - - P - - -",
    " - - - - - - - -",
    " P P P P - P P P",
    " R N B Q K B N R",
]
